src/lib/utils/fetchGitHubContributors.ts

```diff
@@ -0,0 +1,42 @@
+import { join } from "path"
+import { GITHUB_AUTH_HEADERS, GITHUB_COMMITS_URL } from "@/lib/constants"
+import type { Author } from "@/lib/interfaces"
+import type { FileContributorsState } from "@/lib/types"
+
+export const fetchGitHubAuthors = async (
+  relativePath: string
+): Promise<FileContributorsState> => {
+  const url = new URL(GITHUB_COMMITS_URL)
+  const CONTENT_DIR = "/src/content/" // TODO: Overriding with Gatsby repo—Remove and import from constants.ts for production
+  const filePath = join(CONTENT_DIR, relativePath, "index.md")
+  url.searchParams.set("path", filePath)
+
+  try {
+    const uniqueAuthors: Array<Author> = []
+    const response = await fetch(url, GITHUB_AUTH_HEADERS)
+    const commits = await response.json()
+    const authorSet = new Set<string>()
```

Comment:
Can't we make a Set of `Author`s instead?

Reply:
See below... when I did this, it did not treat two matching Author objects as the same, hence stringifying, so they're compared by string value in the Set, then parsed back.